maestrowf/datastructures/core/study.py

```diff
@@ -625,11 +655,15 @@ def _stage(self, dag):
                                 "**********************************",
                                 str(combo))
                     # Compute this step's combination name and workspace.
+                    nickname = None
                     combo_str = combo.get_param_string(self.used_params[step])
+                    # We must encode explicitly to utf-8
+                    # combo_str = combo_str.encode("utf-8")
                     if self._hash_ws:
```

Comment:
Can this be made to turn on automatically if names go beyond a certain length?  might be a nice safety valve for the users so they don't get a bunch of paths that are too long without realizing it?  Or maybe a better check is on the filename lenghts of the shell scripts since those are much shorter than the full path limits: looks like ~250 on linux, path is 4096, but other os's appear to be lower on that.

Reply:
That's an interesting consideration. My main concern with that is some users might use `grep` to find parameter folders and things in post processing. I also don't know a good way to absolutely make sure a user knows that they are being forced into hashed workspaces. This might be worth opening an ideas discussion about and tagging folks who might have something to say on this.

Reply:
well, on the flip side, if the length is too long grepping won't work anyway.   but, not being aggressive with the threshold would be nice as no-hashed makes things easier to navigate with visit and the like too.

definitely worth discussing the how: most of this would happen in background currently right?  makes debug logging as the warning mechanism a bit troublesome

Reply:
Yeah, exactly. The only way that I can think would be to warn the user in the Maestro front end when the number of parameters is large and has a risk of going over. Maybe then ask them a yes/no if they want to hash workspaces. Mind starting an ideas discussion?

Reply:
Sure; I just created a discussion so we can keep going after the PR